src/ncp/hdlc.cpp

```diff
@@ -241,21 +241,23 @@ void Decoder::Decode(const uint8_t *aInBuf, uint16_t aInLength)
                 break;
 
             case kFlagSequence:
-                if (mOutOffset > 2)
+                // We ignore frames which are smaller
+                // than the size of the CRC check.
+                if (mOutOffset > sizeof(uint16_t))
                 {
```

Comment:
Looks good to me as is. Just thinking about this we can also consider not ignoring frames smaller than two and still invoke the error handler and have a print out showing we got them.